OCI: Add squid tests

athos-ribeiro opened the pull request:
This is the initial state of the new squid OCI tests. The first configuration file is just the concatenation of the default Ubuntu configuration files with the docs stripped out. The second, is just a minimal configuration file to ensure we can inject configuration into the image.

athos-ribeiro commented:
This launchpad MP is needed to properly test these changes: https://code.launchpad.net/~canonical-server/ubuntu-docker-images/+git/squid/+merge/408039

sergiodj reviewed (changes requested):
Thanks for the PR, Athos.

I like the tests!  I'm just leaving a few minor comments, but otherwise it LGTM.

For completeness: I was trying to run this using an older version of docker.io (`20.10.2-0ubuntu1~20.04.2`), and was seeing failures with the IPv6 test.  When I updated docker.io to the latest version available on Focal (`20.10.7-0ubuntu1~20.04.1`), the problem disappeared.

athos-ribeiro commented:
Thanks for the review, Sergio!

I addressed all your comments.

Yes, I agree with the approach of pulling the python image before the tests and removing it afterwards!

sergiodj commented:
Awesome, thanks Athos.  +1
